packages/protocol/src/action/server/AttributeResponse.ts

```diff
@@ -248,21 +266,21 @@ export class AttributeResponse<SessionT extends AccessControl.Session = AccessCo
                 }
                 this.#currentEndpoint = endpoint;
                 this.#currentCluster = cluster;
-                this.#currentState = cluster.open(this.#session);
+                this.#currentState = cluster.open(this.session);
             } else if (this.#currentCluster !== cluster) {
                 this.#currentCluster = cluster;
-                this.#currentState = cluster.open(this.#session);
+                this.#currentState = cluster.open(this.session);
             } else if (this.#currentState === undefined) {
-                this.#currentState = cluster.open(this.#session);
+                this.#currentState = cluster.open(this.session);
             }
 
-            // Perform actual read of one attribute
-            this.#addValue(
-                path,
-                this.#currentState[attributeId],
-                cluster.version,
-                this.#currentCluster.type.attributes[attributeId]!.tlv,
+            const value = this.#currentState[attributeId];
+            const version = cluster.version;
+            logger.debug(
```

Comment:
Here's where I think we should have a conditional based on whether debug logging is enabled...  So only do it if debug logging is enabled.  Otherwise we have string formatting, attribute path generation, JSON serialization etc. happening for each attribute that is then unused

Reply:
Done

Reply:
cool